src/groupByKey.ts

```diff
@@ -2,6 +2,15 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(items: object[], key: string): GroupsMap<object> {
+  const result: GroupsMap<object> = {};
+
+  const arrayofValues = items
+    .reduce((sum: string[], el: any) => [...sum, el[key]], []);
+
+  arrayofValues.forEach((value: string) => {
+    result[value] = items.filter((el: any) => el[key] === value);
+  });
```

Comment:
it's too hard an algorithm
can you do it with 1 forEach?
also, you don't need filter
Also, try to avoid "any" type

```suggestion
  const arrayofValues = items
    .reduce((sum: string[], el: any) => [...sum, el[key]], []);

  arrayofValues.forEach((value: string) => {
    result[value] = items.filter((el: any) => el[key] === value);
  });
```